compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -2802,14 +2802,33 @@ abstract class GenJSCode extends plugins.PluginComponent
          *   {name1: arg1, name2: arg2, ... }
          */
 
+        @tailrec
+        def warnIfDuplicatedKey(pairs: List[(js.StringLiteral, js.Tree)], 
+            namesSeen: Set[String] = Set.empty): Unit = {
+          pairs match {
+            case (js.StringLiteral(keyName), _) :: tail =>
+              if (namesSeen(keyName)) {
+                reporter.warning(pos, 
+                    s"Duplicate key in object literal: $keyName. Only the last occurrence is assigned.")
+              }
+              warnIfDuplicatedKey(tail, namesSeen + keyName)
+            case Nil => // Do nothing and finish
+          }
+        }
+
         // Extract first arg to future proof against varargs
         extractFirstArg(genArgs) match {
           // case js.Dynamic.literal("name1" -> ..., "name2" -> ...)
           case (js.StringLiteral("apply"), jse.LitNamed(pairs)) =>
+            warnIfDuplicatedKey(pairs)
             js.JSObjectConstr(pairs)
 
           // case js.Dynamic.literal(x, y)
           case (js.StringLiteral("apply"), tups) =>
+            // Check for duplicated explicit keys
+            val jse.LitNamed(pairs) = tups
```

Comment:
Yes, I made a mistake trying to factor out code. I'll revert to the previous code or try to integrate this case in the `LitNamed` in some coherent way to avoid duplication of code.